data/lib/service/tournament/tournament_service.dart

```diff
@@ -111,4 +118,34 @@ class TournamentService {
       }
     }).handleError((error, stack) => throw AppError.fromError(error, stack));
   }
+
+  Future<List<PlayerKeyStat>> getKeyStats(List<MatchModel> matches) async {
```

Comment:
I think this KeyStats has been mistaken by you, it's not about showing highest run done by each player, participated in tournament but we'll show highest run done in the tournament, similarly other stats like, highest wickets taken by a player, let's look into this area more.